client/client-multi/client-cpp/test/impl/log/SQLiteDBLogStorageTest.cpp

```diff
@@ -108,51 +108,76 @@ BOOST_AUTO_TEST_CASE(GetAllRecordsTest)
 
     KaaUserLogRecord record;
     record.logdata = testLogData;
-    LogRecordPtr serializedRecord(new LogRecord(record));
+    LogRecord serializedRecord(record);
 
     std::size_t recordCount = 5;
     for (std::size_t i = 0; i < recordCount; ++i) {
-        logStorage.addLogRecord(serializedRecord);
+        logStorage.addLogRecord(std::move(serializedRecord));
```

Comment:
After std::move serializedRecord() is in unspecified state. Are you sure that after moving on first iteration it should be used in following?